src/goto-programs/safety_checker.h

```diff
@@ -45,4 +45,39 @@ class safety_checkert:public messaget
   const namespacet &ns;
 };
 
+/// \brief is `r1` a worse result than `r2`?
+inline bool operator<(
+    safety_checkert::resultt &r1,
+    safety_checkert::resultt &r2)
+{
+  switch(r1)
+  {
+  case safety_checkert::resultt::ERROR:
+    return true;
```

Comment:
This is wrong, two things that are `==` must not be `<`. This should return true if r1 is *strictly worse than* r2.